src/components/comment-input.tsx

```diff
@@ -61,10 +77,21 @@ export const CommentInput: React.VoidFunctionComponent = () => {
       <AvatarImage src={userPicture} />
       <TextInput
         placeholder="Добавить комментарий..."
+        isMultiline={true}
+        isDisabled={isLoading}
```

Comment:
```suggestion
        disabled={loading}
```

I’m not sure we’ll need to support single-line comments any time soon, so I’d remove multiline prop as well as the associated complexity. If requirements change in the far future, restoring the prop and adding `StyledTextInput` won’t take much time. Maintaining unused lines of code is generally more expensive.

I’d also suggest to replace `isDisabled` with `disabled` for coherence with `<Button disabled={true} />`. The fact that a prop is boolean does not require us to change the naming rule.`isDisabled` and `isLoading` look more like callable functions, which is not the case here. 

Consequently, we get:

```diff
- const [isLoading, setIsLoading] = React.useState<boolean>(false);
+ const [loading, setLoading] = React.useState<boolean>(false);
```

When `loading` looks too ambiguous in a given context, we can also go with `somethingIsLoading`. This does not necessarily apply here, I’m just giving an example of a variable name that does not start with a verb and is more specific than `loading`.

Reply:
As you can see on the Figma mock https://www.figma.com/file/LkGF1BDzS4dDxMdbx95KmapX/%D0%94%D0%A2%D0%9F?node-id=273%3A2816 we have a sindle line input for the search field, so it's already require to have both multiline and single line text inputs. The only question if this should be separated compoenent or just a prop. For me using a prop is more flexible

Reply:
As for me isLoading looks much more boolean variable, comparing to loading. Ok, I'll rename them both.

Reply:
Yeah I agree that `isDisabled` and `isLoading` may feel a bit more intuitive. If we apply a special rule for booleans, why not to other types? That would give us `emailString`, `dataObject` and so on, which look quite odd. So if there are no rules for other primitive types, why have one for Boolean?

The two rules I tend to apply for variable names are:

- Is it callable? If yes, start with a verb `isLoading()` vs `loading`
- Can be looped over? If yes, use plural: `things` vs `thingLookup` / `thingById` for lookups

Not using `is` for booleans is common in third-party libs, so we have a more natural naming integration.

---

I would say that thinking about search is a bit too early at this stage. Introducing a single-line input may become a wasted effort if we decide to introduce a third-party component library like [MUI](https://mui.com) by the time we get there. Besides, we’ll probably need to refactor the current implementation of single-line input because it does not support submit on enter, the icon on the right and so on. Thus, future-proofing ends up more expensive than thinking only about what we need right now.

A lightweight approach may feel like bad engineering because developers are taught to think long-term, keep things DRY and avoid refactoring. This is justified in some languages and types of software. With React and TypeScript, refactoring is cheap because the architecture of the environment we are in is designed for making changes. Adding and removing lines of code does not cost much because our building blocks (components and hooks) are both flexible and robust. If an error is added during refactoring, it will be likely spotted by linting and so the problem won’t get to prod. On the contrary, trying to introduce conditions before we need them slows down feature shipping and potentially increases tech debt.

Reply:
Sure, I like to have the only actualized code as well.